Changes required for branding, hide of pop ups

sppanday opened the pull request:
This pull request consists of branding configuration:

Config values for default color codes
Config values for hiding pre login screen
Config values for hiding post login prompt
Hide "Liberate your communication" message once user is authenticated

===============================================
Images and theme style files need to be replaced with docker volume with Linux command in order to reflect changes from =========================================

aether-host-files/common/element-web/images
aether-host-files/common/element-web/styles

sppanday committed:
changes for logo, hide popup, skip prelogin prompt

sppanday committed:
login prompt checks if e2e is enabled from server

sppanday committed:
secondary logo - increased gap

jrschremmer commented on src/DeviceListener.ts:
```diff
@@ -282,9 +284,9 @@ export default class DeviceListener {
             }
         }
-
+        const showSecurityKeyBackupPrompt = () => SettingsStore.getValue(UIFeature.ShowSecurityKeyBackupPrompt);
```
Can the "() =>" be removed as I think it just adds confusion?  In other similar code you don't have it e.g.
const showLiberateToCommunicateText = SettingsStore.getValue(UIFeature.ShowLiberateToCommunicateText);

sppanday replied:
This has been actioned now.  You are right my innovation wasnt required there its just a single config pull so this has been actioned.

jrschremmer commented on src/components/structures/HomePage.tsx:
```diff
@@ -94,7 +96,7 @@ const UserWelcomeTop = () => {
     const config = SdkConfig.get();
     const pageUrl = getHomePageUrl(config);
-
+    const showLiberateToCommunicateText = SettingsStore.getValue(UIFeature.ShowLiberateToCommunicateText);
```
Please rename ShowLiberateToCommunicateText to ShowLiberateYourCommunicationText to match the code

sppanday replied:
Change request/suggestion has been actioned now.

jrschremmer commented on src/components/structures/MatrixChat.tsx:
```diff
@@ -394,7 +395,18 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
+        // This can be used to check if encryption is enabled/disabled in home-server.
+        const encryptionIsEnabled = () => {
+            if (!SettingsStore.getValue(UIFeature.LookForWellKnownConfigFromHomeServer)) {
+                return null;
```
I don't understand return null ... perhaps u meant return true?
As discussed in the other PR for the matrix-js-sdk, if the UIFeature.LookForWellKnownConfigFromHomeServer is moved into that code, then this if statement may no longer be necessary.

sppanday replied:
This "if condition" has now become redundant when check is done in autodiscovery.js so i have removed it now after implementing your suggestion. I think I was trying to tell code to do nothing if config is not found. I am not sure if i was right with that null there though.

jrschremmer commented on src/components/structures/MatrixChat.tsx:
```diff
@@ -394,7 +395,18 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
+            if (!SettingsStore.getValue(UIFeature.LookForWellKnownConfigFromHomeServer)) {
+                return null;
+            }
+            console.log("well-known config has default e2e value set to::: ", getE2EEWellKnown().default);
```
Please save getE2EEWellKnown().default into a local variable instead of calling it twice.

sppanday replied:
Yay, thats great idea, I did not realise, fixed now.

jrschremmer commented on src/components/structures/MatrixChat.tsx:
```diff
@@ -394,7 +395,18 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
+                return null;
+            }
+            console.log("well-known config has default e2e value set to::: ", getE2EEWellKnown().default);
+            return (getE2EEWellKnown().default === true) ? true : false;
```
Please remove the redundant ? true : false

sppanday replied:
This has been actioned in new commit.

sppanday committed:
cleaned up code based on #12

jrschremmer committed:
Code review:
Rename ShowLiberateToCommunicateText to ShowLiberateYourCommunicationText;
Centralised the common logic in MatrixChat.tsx and createRoom.ts into WellKnownUtils.isE2EEEnabledInWellKnown;
Default LookForWellKnownConfigFromHomeServer to false, so the default element behaviour is maintained;

jrschremmer committed:
In WellKnownUtils.ts changed isE2EEEnabledInWellKnown to lookup the new enabled property and isE2EEDefaultedOnInWellKnown to lookup the existing default property
